content/documentation/pages/4-batch-developer-guides/3-troubleshooting/2-debugging-scdf-tasks.md

```diff
@@ -4,5 +4,146 @@ title: 'Debugging Batch applications deployed by Data Flow'
+	B --> |Yes| C(Resolve exceptions)
+	B --> |No| D(Exceptions in app logs?)
+	D -->|Yes| E(Resolve exceptions)
+	D -->|No| F(Raise log levels)
+    F --> G(Enable debug)
+```
+
+Application log files can be inspected on a per application basis.
+To aggregate logs from all applications into one, the deployer property `inheritLogging=true` can be set.
+See
+[Log Redirect](http://docs.spring.io/spring-cloud-dataflow/docs/current/reference/htmlsingle/#_log_redirect)
+for more information and [Deployment Logs](http://docs.spring.io/spring-cloud-dataflow/docs/current/reference/htmlsingle/#troubleshooting-deployment-logs) for enabling more log output.
+
+Debugging applications via JDWP can be accomplished by setting the deployer property `debugPort`.
+See [Remote Debugging](http://docs.spring.io/spring-cloud-dataflow/docs/current/reference/htmlsingle/#_remote_debugging) for more information.
+
+# Docker Compose - Startup
+
+```mermaid
+graph TD;
+    A{DATAFLOW_VERSION defined?} --> |No| A
+    A --> |Yes| B{SKIPPER_VERSION defined?}
+	B --> |No| B
+```
+
+The environment variables `DATAFLOW_VERSION` and `SKIPPER_VERSION` must be available in the current terminal environment via `export` or prefixing the `docker-compose` command.
+See [Starting Docker Compose](http://docs.spring.io/spring-cloud-dataflow/docs/current/reference/htmlsingle/#getting-started-local-deploying-spring-cloud-dataflow-docker-starting) for more information.
+
+# Docker Compose - Runtime
+
+```mermaid
+graph TD;
+    A{Low Resources?} --> |Yes| B(Increase Docker Resources)
+    A --> |No| C{Exceptions in app logs?}
+	C --> |Yes| D(Resolve exceptions)
+```
+
+By default, the amount of memory allocated to Docker may be too low.
+The recommended amount of memory to allocate is 8GB.
+The command `docker stats` can provide useful information into resource usage.
+If applications are failing to deploy due to resource constraints, increase resource allocations.
+Consult the [Docker documentation](https://docs.docker.com/) for your platform.
+
+As tasks are launched via SCDF, applications that are part of that task definition will be launched as Java processes on the SCDF container.
+For every part of a task definition, an application is launched.
+The overall resource allocation (memory, CPU, etc) provided to Docker should account for the number of deployed applications.
+
+# Cloud Foundry - Startup failures
+
+```mermaid
+graph TD;
+    A(Verify Manifest) --> B(Verify environment variables);
+    B --> C(Service instances running?)
+    C --> |No| C
+    C --> |Yes| D(Exceptions in SCDF logs?)
+    D --> |No| E(View logs and resolve errors)
+```
+
+# Cloud Foundry - Application failures
+
+```mermaid
+graph TD;
+    A{Applications Started?} --> |No| B{Errors in SCDF logs?}
+    A{Applications Started?} --> |No| F{Errors in app logs?}
+    B --> C(Resolve)
+	F --> C
+```
+
+When debugging deployment issues, raising deployer and Cloud Foundry releated log levels may be useful.
```

Comment:
typo: "releated" (perhaps also a typo in the stream guide since we copied content)